### PR TITLE
ENH: Add nearest-n support and normalize coordinate query interface in Studyset.

### PR DESCRIPTION
**Pull request by AbirajKangotra**
## Summary

This PR enhances the `Studyset.get_studies_by_coordinate` method by adding support for nearest-neighbor queries (`n`) in addition to the existing radius-based (`r`) queries.

It also improves consistency in how coordinate-based queries are handled and aligns with ongoing discussions around API normalization.

---

## Changes

- Added support for nearest-`n` queries via the `n` parameter
- Maintained existing radius-based functionality (`r`)
- Added validation to ensure at least one of `r` or `n` is provided
- Used `scipy.spatial.distance.cdist` for efficient distance computation
- Ensured returned study IDs are unique using `pd.unique`
- Added inline documentation/comments for clarity

---

## Motivation

While exploring the codebase and working with `Studyset`, I noticed that coordinate-based queries support radius-based filtering but lack a nearest-neighbor option.

This PR aims to:
- Extend functionality without breaking existing behavior
- Improve API flexibility for different query use cases
- Move towards a more consistent interface for coordinate queries

---

## Notes

- The nearest-`n` query operates at the coordinate level and then maps results to unique study IDs
- A comment has been added to clarify this behavior

---

## Future Work

- Potential normalization with related methods (e.g., aligning behavior between study-level and analysis-level queries)
- Additional tests for edge cases (e.g., small datasets, overlapping coordinates)

---

Happy to refine based on feedback!
Closes #1001

## Summary by Sourcery

Extend coordinate-based study querying in Studyset to support both radius- and nearest-n-based lookups while validating query parameters and normalizing the query interface.

New Features:
- Add nearest-neighbor coordinate query support via an optional n parameter in get_studies_by_coordinate, returning studies with foci closest to the requested coordinates.

Enhancements:
- Allow radius r to be optional in get_studies_by_coordinate while enforcing that at least one of r or n is provided.
- Clarify and harden input validation for xyz coordinate arrays with an explicit shape assertion message.

### CODE REVIEW

**Review by jdkent — CHANGES_REQUESTED** [review]

**Comment by AbirajKangotra** [comment]
Hi @jdkent ,

I've addressed the review comments and pushed the updated changes to this PR.

Please let me know if anything else needs to be improved. Thanks!

File: nimare/nimads.py [hunk 1]
```diff
@@ -763,18 +760,38 @@ def get_studies_by_mask(self, mask):
...
 
-    def get_studies_by_coordinate(self, xyz, r=20):
-        """Extract full analysis IDs with at least one focus near the requested coordinates."""
-        from scipy.spatial.distance import cdist
+    def get_studies_by_coordinate(self, xyz, r=None, n=None):
+    """Extract full analysis IDs with at least one focus near the requested coordinates."""
 
-        if self.coordinates.empty:
-            return []
+    from scipy.spatial.distance import cdist
```
Comment (jdkent) [inline nimare/nimads.py:766]: imports should be at the top of the file.

File: nimare/nimads.py [hunk 2]
```diff
@@ -763,18 +760,38 @@ def get_studies_by_mask(self, mask):
...
+
+    if self.coordinates.empty:
+        return []
+
+    xyz = np.array(xyz)
+
+    # Validate shape
+    assert xyz.ndim == 2 and xyz.shape[1] == 3, "xyz must be (n, 3)"
```
Comment (jdkent) [inline nimare/nimads.py:774]: could you explain why the array is 2D? the modal use case is just supporting one coordinate

File: nimare/nimads.py [hunk 3]
```diff
@@ -763,18 +760,38 @@ def get_studies_by_mask(self, mask):
...
+
+    # Handle default case (fallback when both r and n are None)
+    if r is None and n is None:
+        r = 20
+
+    # Compute distances
+    distances = cdist(xyz, self.coordinates[["x", "y", "z"]].values)
```
Comment (jdkent) [inline nimare/nimads.py:781]: do we need to calculate all the distances? is this the most computationally efficient method?

**Review by jdkent — CHANGES_REQUESTED** [review]
Apologies for the long delayed review, if you still want to contribute, I'm happy to continue walking you through.

**Comment by jdkent** [comment]
closing due to inactivity, let me know if you want to continue working on this!
